utils/node_local_test.py

```diff
@@ -5772,6 +5800,48 @@ def test_alloc_fail(server, conf):
     return rc
 
 
+def server_fi(args):
+    """Run the server under fault injection.
+
+    Start the server, create a container, enable periodic failing of D_ALLOC() and then perform
+    I/O.  At some point this could be extended to checking the client also behaves properly but
+    for now just check the server logs.
+
+    This is not run in CI yet so needs to run manually.  As it's probabilistic then it can be
+    expected to find more issues based on how often it's run so it is not suitable for PRs
+    but should be run for long periods of time.
+    """
+    conf = load_conf(args)
+
+    wf = WarningsFactory('nlt-errors.json', post_error=True, check='Server FI testing')
+
+    args.dfuse_debug = 'INFO'
+    args.client_debug = 'INFO'
+    args.memcheck = 'no'
+
+    conf.set_wf(wf)
+    conf.set_args(args)
+    setup_log_test(conf)
+
+    with DaosServer(conf, wf=wf, test_class='server-fi', enable_fi=True) as server:
+
+        pool = server.get_test_pool_obj()
+        cont = create_cont(conf, pool=pool, ctype='POSIX', label='server_test')
+
+        # Instruct the server to fail a % of allocations.
+        server.set_fi(probability=5)
+
+        for idx in range(100):
+            server.run_daos_client_cmd_pil4dfs(
+                ['touch', f'file.{idx}'], container=cont, check=False, report=False)
+            server.run_daos_client_cmd_pil4dfs(
+                ['rm', '-f', f'file.{idx}'], container=cont, check=False, report=False)
+
+        # Turn off fault injection again to assist in server shutdown.
+        server.set_fi(probability=0)
+        server.set_fi(probability=0)
```

Comment:
We probably don't, we might want to always do it when it's been enabled so it could go in the DaosServer() logic however at some point we should start checking allocation failures on shutdown so better to keep it here and not complicate the DaosServer() object I think.